src/main/java/interfaces/AppATM.java

```diff
@@ -0,0 +1,14 @@
+package interfaces;
+
+public class AppATM {
+    public static void main(String[] args) {
+
+        ATMImp1 useAtm = new ATMImp1();
```

Comment:
If you use implementation of interface or abstract class, you need to declare this class as an interface type. In this case this line should look like this: 

CashMachine atm = new AtmImpl(); ---> at the end is (small L not 1)

example:

lets imagine that you have two implementation of interface CashMachine,
1. AtmCashMachine
2. PlanetCashMachine

and you use a payment service to realize payOut, but it depends on cashMachineType. (in this example this cashMachineType you got from an external service), this class is only proceeding request.

public class PaymentService {

    private CashMachine atm = new AtmCashMachine();
    private CashMachine planet = new PlanetCashMachine();
    private String ATM_TYPE = "ATM"

    public void processPayOut(int amount, String cashMachineType) {

    CashMachine processor = (ATM_TYPE.equals(cashMachineType)) ? atm : planet;
    processor.payOut();

    }
}

Reply:
so if I understand correctly, because of the interface, we can use the payOut() method dynamically, depending on what class of this Interface we have? 

If we tried to do the same thing without Interface, we would have to define two separate methods in two separate classes atm and planet for example:

payOutATM
payOutPlanet 

and use some kind of if statements to make this work in a similar way?

if(ATM_TYPE == planet) {
etc..?